### DISCUSSION TITLE
Refresh token not saved in db Account signin

### DISCUSSION THREAD

**Discussion by markbjs** [nextauthjs/next-auth#12313] [Help] [answered]
hello i am using 

```
    "next": "^15.0.2",
    "next-auth": "^5.0.0-beta.25",
```
with
```
    "@upstash/redis": "^1.34.3",
    "@auth/upstash-redis-adapter": "^2.7.4",
```

using database strategy  and EntraID provider using

import MicrosoftEntraID from "next-auth/providers/microsoft-entra-id";

```
export const { handlers, signIn, signOut, auth } = NextAuth({
  ...authConfig,
  providers: [
    MicrosoftEntraID({
      authorization: {
        params: {
          scope: `....`,
        },
      },
      clientId: process.env.ENTRA_AD_CLIENT_ID,
      clientSecret: process.env.ENTRA_AD_CLIENT_SECRET,
      issuer: process.env.ENTRA_ID_ISSUER,
    }),
  ],
```

we have account saved in DB with no refresh_token:
```
{
  "access_token": ...,
  "id_token": ...,
  "expires_at": 1732790111,
  "scope": "...",
  "token_type": "bearer",
  "providerAccountId": "...",
  "provider": "microsoft-entra-id",
  "type": "oidc",
  "userId": "...",
  "id": "microsoft-entra-id:..."
}
```

is this a bug that we have no refresh token saved, 

any help appreciated 

thanks

**Comment by markbjs** [comment, marked as answer]
Sorry found solution need to request offline_access graph scope to get the refresh_token returned

**Reply by Christoffer-Toft** [reply]
@markbjs  did you have to add that permission specifically on the enterprise app as well for it to work? MIgrating from azure-ad to microsoft-entra-id and havent had this problem before
